### PR TITLE
(DOCSP 17936) Find TS

### PR DESCRIPTION
**Pull request by biniona-mongodb**
## Pull Request Info
Add TS Variant for Find Multiple Usage Example

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-17963

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-17936-Find-Multiple/usage-examples/find/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### CODE REVIEW

File: source/code-snippets/usage-examples/find.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { MongoClient } from "mongodb";
+
+// Replace the uri string with your MongoDB deployment's connection string.
+const uri =
+  "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
+
+const client = new MongoClient(uri);
+
+interface Movies {
```
Comment (cbush) [inline source/code-snippets/usage-examples/find.ts:9]: Why plural? Doesn't this interface represent one Movie?

File: source/code-snippets/usage-examples/find.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  title: string;
+  imdb: object;
+  runtime: number;
```
Comment (cbush) [inline source/code-snippets/usage-examples/find.ts:12]: Runtime what? Minutes, seconds...?

File: source/code-snippets/usage-examples/find.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+// Replace the uri string with your MongoDB deployment's connection string.
+const uri =
+  "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
+
+const client = new MongoClient(uri);
+
+interface Movies {
+  title: string;
+  imdb: object;
```
Comment (cbush) [inline source/code-snippets/usage-examples/find.ts:11]: This should have been a linter warning - avoid "object". Did you mean `Record<string, unknown>`? (Or better yet, something more specific... imdb url?)

>Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys.

File: source/code-snippets/usage-examples/find.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    const cursor = movies.find<Movies>(
+      { runtime: { $lt: 15 } },
+      {
+        sort: { title: 1 },
+        projection: { _id: 0, title: 1, imdb: 1 },
+      }
+    );
+
+    if ((await cursor.count()) === 0) {
+      console.log("No documents found!");
```
Comment (cbush) [inline source/code-snippets/usage-examples/find.ts:31]: Maybe console.warn?

**Review by cbush — APPROVED** [review]
A few stronger recommendations, looks good assuming they are fixed.

File: source/code-snippets/usage-examples/find.ts (hunk 1 above)
Comment (biniona-mongodb) [reply source/code-snippets/usage-examples/find.ts:9]: Changed to `Movie`

File: source/code-snippets/usage-examples/find.ts (hunk 3 above)
Comment (biniona-mongodb) [reply source/code-snippets/usage-examples/find.ts:11]: Woah cool issue, definitely don't understand everything about why using the object type is discouraged, but the length and depth of that discussion says something about it being hard to use. imdb fields are small so I'll just make an interface for them.

**Comment by biniona-mongodb** [comment]
Hi @mongodb/dbx-node, could I have a review for this TypeScript usage Example?

**Review by dariakp — APPROVED** [review]
LGTM

**Pull request merged**